### PR TITLE
use resolvconf package from c/common/libnetwork

### PR DESCRIPTION
**Pull request by Luap99**
Podman and Buildah should use the same code the generate the resolv.conf
file. This mostly moved the podman code into c/common and created a
better API for it so buildah can use it as well.

[NO NEW TESTS NEEDED] All existing tests should continue to pass.

Fixes https://github.com/containers/podman/issues/13599 (There is no way to test this in CI without breaking the
hosts resolv.conf)

#### Does this PR introduce a user-facing change?

```release-note
Podman will now check for nameservers in /run/NetworkManager/no-stub-resolv.conf if the /etc/resolv.conf file only contains a localhost server.
```

### CODE REVIEW

**Comment by Luap99** [comment]
@mheon @rhatdan PTAL

**Comment by mheon** [comment]
LGTM

**Comment by rhatdan** [comment]
Tests are failing?

File: libpod/container_internal_linux.go [hunk 1]
```diff
@@ -2385,38 +2329,34 @@ func (c *Container) generateResolvConf() error {
...
-	if _, err = resolvconf.Build(destPath, nameservers, search, options); err != nil {
+	if err := resolvconf.New(&resolvconf.Params{
+		Path:            destPath,
+		Namespaces:      c.config.Spec.Linux.Namespaces,
+		IPv6Enabled:     ipv6,
+		KeepHostServers: keepHostServers,
+		Nameservers:     nameservers,
+		Searches:        search,
+		Options:         options,
+	}); err != nil {
```
Comment (TomSweeneyRedHat) [inline libpod/container_internal_linux.go:2355]: It would be nice to have these in alpha order....

Comment (Luap99) [reply libpod/container_internal_linux.go:2355]: Why? I feel like this order makes the most sense to me.

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**